### ISSUE TITLE
helm3 --dry-run does not error on invalid policyType 

### ISSUE THREAD

**Issue by teng1** [helm/helm#6485]
Hi, I noticed the following passed the dry run with no issues but fails on apply due to a typo in policyType

```
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  name: {{ template "..fullname" . }}
  labels:
{{ include "..labels" . | indent 4 }}
spec:  
  podSelector:
    matchLabels:
      name: {{ include "..name" . }}
      instance: {{ .Release.Name }}
  policyTypes:
  - Ingres
  ingress:
  - from:
    - podSelector:
        matchLabels:
          name: {{ include "..name" . }}
          instance: {{ .Release.Name }}         
    ports:
    - protocol: TCP
      port: 3128
```
 (Ingres rather than Ingress) would it be expected for the --dry-run to catch that?

Thanks

**Comment by thomastaylor312** [comment]
Hey @teng1, thanks for reporting this! So this is a known issue with the beta right now. When we shuffled around all of the code, the open API validation code was removed. @adamreese is working on a PR right now to bring that functionality back in, which should allow dry run to catch it properly.

**Comment by bacongobbler** [comment]
fixed in #6599
